### PR TITLE
Validate google spreadsheet url

### PR DESCRIPTION
**Pull request by carvil**
Validate the URL provided to `TaggingSpreadsheet` models in order to make sure only valid Google Spreadsheet URLs are given by the user.

There are 3 validations:
- the hostname (it should be `docs.google.com`)
- the path (it needs to match `/spreadsheets/d/<a key>/pub`
- the params (it needs to include a Google ID param called `gid`)

This PR also adds a factory for building `TaggingSpreadsheet` objects in the tests with the correct URL structure.

Trello: https://trello.com/c/AGgxBxcJ/87-validate-the-data-given-to-the-tag-importer-it-should-be-a-google-spreadsheet-url-currently-it-accepts-anything

Part of: https://trello.com/c/7tcrUgcz/66-initial-round-of-improvements-to-error-handling-in-tag-importer

### CODE REVIEW

File: app/validators/google_url_validator.rb [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+    end
+  end
+
+  def validate_path(uri, record)
+    unless uri.path =~ %r{spreadsheets\/d\/.+\/pub}
+      record.errors[:url] << I18n.t('errors.invalid_path')
+    end
+  end
+
+  def validate_parameters(uri, record)
```
Comment (mobaig) [inline app/validators/google_url_validator.rb]: Do you think it's worth validating somewhere that `output=tsv` in the params? The CSV parsing logic in `app/services/tag_importer/fetch_remote_data.rb` expects tab-separated values.

Comment (carvil) [inline app/validators/google_url_validator.rb]: Ah, well spotted @mobaig I will add that! Thanks

Comment (carvil) [inline app/validators/google_url_validator.rb]: @mobaig : I added that validation below, is that correct?

**Comment by mobaig** [comment]
👍

**Pull request merged**